### DISCUSSION TITLE
New log rendering terminal

### DISCUSSION THREAD

**Discussion by paladium** [NativeCI/Issues#20] [Ideas]
I am currently starting to work on new terminal for rendering logs, similar to what github actions and travis ci have. Are there any specific features, you would like to me add besides the minimum:
- Collapsible sections for build, linting, tests etc.
- Link to line number
- Some emojis and colors for statuses

**Comment by paladium** [comment]
@gucio321 do you have any additional comments?

**Reply by gucio321** [reply]
very cool terminal, no questions for now

**Reply by gucio321** [reply]
oh, one thing, you can indroduce some interpretation of golangci-lint alerts. atm it looks like:
```console
 level=error msg="Running error: unknown linters: 'forcetypeassert,importas,nilerr,promlinter,wastedassign', run 'golangci-lint linters' to see the list of supported linters"
16:55:35 
```

**Reply by paladium** [reply]
You mean to highlight warnings or display the output of the linting within the code?

**Comment by paladium** [comment]
@gucio321 the new terminal for log output is out. It already has collapsible section for git clone to simplify overview of the build, the next step would be to add the same for other steps, like dependency installation, as it gets quite huge when for example libgtk-3-dev is being installed.

**Comment by paladium** [comment]
@gucio321 I have released a new build strategy, it now folds different sections of the build like linting, testing and building. Could you try to build your projects to see if you run into any issues?
![image](https://user-images.githubusercontent.com/8057926/126517202-19cd3566-6af0-4918-893a-295030953919.png)

**Reply by gucio321** [reply]
jsure, just doing that

**Reply by gucio321** [reply]
hmm, I found the following problem: it is unable to turn off auto scrolling.
when steps going on, terminal automatically scrolls down the output, but when the user want to see some logs before, he scrolls up (using mouse). these time, auto scrolling should be turned off. when user scroll to the bottom of logs, auto-scroll should be tourned on again.

**Reply by gucio321** [reply]
also, i think, im unable to see some logs/sections
![image](https://user-images.githubusercontent.com/73652197/126545950-2db07298-934a-4d68-8fb8-ba8093cddebe.png)
